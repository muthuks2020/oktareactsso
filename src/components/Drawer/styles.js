import { makeStyles } from "@material-ui/core/styles";

const drawerWidth = 240;

export const useDrawerStyles = makeStyles({
  drawer: {
    width: drawerWidth,
    flexShrink: 0,
  },
  drawerPaper: {
    width: drawerWidth,
  },
  link: {
    textDecoration: "none",
  },
});