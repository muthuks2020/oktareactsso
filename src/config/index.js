export { default as securityConfig } from './security'
export { default as securedRoutes } from './routes'