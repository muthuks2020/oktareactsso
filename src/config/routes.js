// imports
import React from "react";
import { Lock } from "@material-ui/icons";
import { Home, Protected } from "pages";

export default [
  {
    icon: () => <Lock />,
    component: () => <Home />,
    exact: true,
    title: "Home",
    path: "/",
  },
  {
    icon: () => <Lock />,
    component: () => <Protected />,
    exact: false,
    title: "Protected",
    path: "/protected",
  },
];
