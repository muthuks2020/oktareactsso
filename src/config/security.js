const {
  REACT_APP_CLIENT_ID: clientId,
  REACT_APP_DOMAIN,
  HOST = "localhost",
  PORT = 3000,
  HTTPS = false,
} = process.env;

const issuer = REACT_APP_DOMAIN + "/oauth2/default";

const PROTOCOL = HTTPS ? "https" : "http";

const redirectUri = `${PROTOCOL}://${HOST}:${PORT}/implicit/callback`;

const scopes = ["openid", "profile", "email"];

export default {
  clientId,
  issuer,
  pkce: true,
  redirectUri,
  scopes,
};
