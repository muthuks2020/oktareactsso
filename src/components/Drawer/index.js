import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { useOktaAuth } from "@okta/okta-react";
import {
  Drawer as MuiDrawer,
  Divider,
  ListItemIcon,
  ListItemText,
  List,
  ListItem,
} from "@material-ui/core";
import { PowerSettingsNew } from "@material-ui/icons";
import { securedRoutes } from "config";
import { useDrawerStyles } from "./styles";

const Drawer = props => {
  const classes = useDrawerStyles();
  const { authService } = useOktaAuth();

  const logout = async () => authService.logout("/");

  return (
    <MuiDrawer
      className={classes.drawer}
      variant={props.variant}
      classes={{
        paper: classes.drawerPaper,
      }}
      anchor={props.anchor}
    >
      <List>
        {securedRoutes.map((route) => (
          <ListItem button component={Link} key={route.path} to={route.path}>
            <ListItemIcon>{<route.icon />}</ListItemIcon>
            <ListItemText primary={route.title} />
          </ListItem>
        ))}
      </List>
      <Divider />
      <ListItem button onClick={() => logout("/")}>
        <ListItemIcon>
          <PowerSettingsNew />
        </ListItemIcon>
        <ListItemText primary="Logout" />
      </ListItem>
    </MuiDrawer>
  );
};

Drawer.propTypes = {
  anchor: PropTypes.string,
  variant: PropTypes.string,
};

Drawer.defaultProps = {
  anchor: "left",
  variant: "permanent",
};

export default Drawer;
