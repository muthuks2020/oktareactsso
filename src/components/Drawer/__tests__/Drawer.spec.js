import React from "react";
import { mount } from "enzyme";
import Drawer from "..";
import { Security } from "@okta/okta-react";
import { securityConfig } from "config";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";

describe("<Drawer />", () => {
  let authService;
  let authState;
  let mockProps;
  beforeEach(() => {
    authState = {
      isPending: true,
    };
    authService = {
      on: jest.fn(),
      updateAuthState: jest.fn(),
      getAuthState: jest.fn().mockImplementation(() => authState),
      handleAuthentication: jest.fn(),
    };
    mockProps = { authService, ...securityConfig };
  });

  test("should render component with default props", () => {
    const component = mount(
      <MemoryRouter>
        <Security {...mockProps}>
          <Drawer />
        </Security>
      </MemoryRouter>
    );
    expect(component.find('Drawer').prop('anchor')).toEqual('left');
    expect(component.find('Drawer').prop('variant')).toEqual('permanent');
  });

  test("should render to snapshot", () => {
    let component = renderer.create(
      <MemoryRouter>
        <Security {...mockProps}>
          <Drawer />
        </Security>
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
