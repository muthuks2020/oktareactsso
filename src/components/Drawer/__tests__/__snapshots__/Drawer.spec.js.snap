// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer /> should render to snapshot 1`] = `
<div
  className="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-1"
>
  <div
    className="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-2 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
  >
    <ul
      className="MuiList-root MuiList-padding"
    >
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        href="/"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Home
          </span>
        </div>
      </a>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        href="/protected"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Protected
          </span>
        </div>
      </a>
    </ul>
    <hr
      className="MuiDivider-root"
    />
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M13 3h-2v10h2V3zm4.83 2.17l-1.42 1.42C17.99 7.86 19 9.81 19 12c0 3.87-3.13 7-7 7s-7-3.13-7-7c0-2.19 1.01-4.14 2.58-5.42L6.17 5.17C4.23 6.82 3 9.26 3 12c0 4.97 4.03 9 9 9s9-4.03 9-9c0-2.74-1.23-5.18-3.17-6.83z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Logout
        </span>
      </div>
    </div>
  </div>
</div>
`;
